### PR TITLE
Update OpenRCT2 Plugin API Type Definitions to v0.4.x

### PR DESCRIPTION
**Pull request by tubbo**
The types for OpenRCT2 are out-of-date, and quite a few breaking  changes to the
plugin API have occurred since this was first contributed. This PR updates the
type definitions for OpenRCT2 to match the `v0.4.x` series, and I'm starting to
think about how to best automate this since it's really just copy-pasting from
an existing `.d.ts` file on GitHub (just not released to NPM, so it's difficult
to work with).
                                                                 
## Summary of Changes                                            
                                                                 
- add new type definitions                                       
- update examples and tests to match new api requirements        
- bump version to `v0.4.x`                                         
- disable lint rules for array types to preserve `.d.ts` format

### CODE REVIEW

File: types/openrct2/package.json [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-            "name": "Ted John",
-            "githubUsername": "IntelOrca"
-        }
-    ]
+    {
+      "name": "Ted John",
+      "githubUsername": "IntelOrca"
+    },
+    {
+      "name": "Tom Scott",
```
Comment (tubbo) [inline types/openrct2/package.json:21]: sorry about the weird diff here. @IntelOrca i added myself as an owner so i could help out releasing new versions and contributing the `.d.ts` myself...if this is not cool let me know and i can remove.

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/openrct2/.eslintrc.json
1. types/openrct2/package.json
1. types/openrct2/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against fa19f9b1aedf5fa4dcbb29aaf31d371e46a9616d
</p>

File: types/openrct2/test/openrct2-tests.example.ts [hunk 2]
```diff
@@ -136,8 +136,8 @@ function open() {
                     ["", ""],
                     sep("VehicleObject"),
                     ["Rotation frame mask:", vehicleObject.rotationFrameMask.toString()],
-                    ["Num. of vertical frames:", vehicleObject.numVerticalFrames.toString()],
-                    ["Num. of horizontal frames:", vehicleObject.numHorizontalFrames.toString()],
+                    ["Num. of vertical frames:", vehicleObject.numVerticalFramesOverride.toString()],
+                    ["Num. of horizontal (base) frames:", vehicleObject.baseNumFrames.toString()],
```
Comment (tubbo) [inline types/openrct2/test/openrct2-tests.example.ts:140]: I'm 99% sure this is the intended replacement for `numHorizontalFrames` but I'm not really sure? Did this just get removed or was this a good guess? :)

Comment (IntelOrca) [reply types/openrct2/test/openrct2-tests.example.ts:140]: Not sure, you can just remove them if you like. This example file is not exhaustively testing every API and every property anyway.

**Review by IntelOrca — APPROVED** [review]

**Review by rbuckton — APPROVED** [review]

**Comment by tubbo** [comment]
Ready to merge

**Pull request merged**